scripts/pdpModule.js

```diff
@@ -0,0 +1,173 @@
+    const menuItemsList = fetchMenuList();
+    const cartItemsList = fetchCartList();
+    const itemsByCategoryMap = createItemsByCategoryMap(menuItemsList);
+
+    function display() {
+        _displayCategories("recommended");
+        _displayMenu("recommended");
+        _displayCart();
+    }
+
+    function _createCategoryItem(element,category,categoriesItemListEl){
+        const categoriesItemEl = createDomNode("li",undefined,element.displayName);
+        if (element.id == category) {
+            categoriesItemEl.classList.add("highlighted");
+        }
+        categoriesItemEl.id = element.id;
+        categoriesItemListEl.append(categoriesItemEl);
+    }
+    
+    function _displayCategories(category) {
+        const categoriesContainer = document.querySelector(".categories");
+        const categoriesItemListEl = createDomNode("ul");
+        categoriesItemsList.forEach(element => _createCategoryItem(element,category,categoriesItemListEl));
+        categoriesContainer.append(categoriesItemListEl);
+    }
+
+    function _createMenuItem(category, index) {
+        const menuItem = getItem(itemsByCategoryMap, category, index);
+        const menuItemEl = createDomNode("li",["dish"]);
+        const vegMarkEl = createDomNode("img");
+        vegMarkEl.setAttribute("src", "./images/vegMark.png");
+        const dishNameEl = createDomNode("p",["dish-name"],menuItem.displayName);
+        const priceEl = createDomNode("p",["price"],`₹ ${menuItem.price}`);
+        menuItemEl.append(vegMarkEl, dishNameEl, priceEl);
+        return menuItemEl;
+    }
+
+    function _createMenuList(category) {
+        const menuListEl = createDomNode("ul");
+        for (let i = 0; i < getItemListByCategory(itemsByCategoryMap, category).length; i++) {
+            menuListEl.append(_createMenuItem(category, i));
+        }
+        return menuListEl;
+    }
+
+    function _displayMenu(category) {
+        const menuLength = getItemListByCategory(itemsByCategoryMap, category) ? getItemListByCategory(itemsByCategoryMap, category).length : 0;
+        const menuHeading = categoryNameById(categoriesItemsList, category);
+        const menuList = _createMenuList(category);
+        const menuContainer = document.querySelector(".menu");
+        const menuheadingEl = createDomNode("h2",undefined,menuHeading);
+        const menuSubheadingEl = createDomNode("p",undefined,`${menuLength} Items`);
+        menuContainer.append(menuheadingEl, menuSubheadingEl, menuList);
+    }
+
+    function _createCartList(element,cartItemsEl){
+        const cartItemEl = createDomNode("li", undefined, element.name)
+        cartItemEl.id = element.id;
+        cartItemsEl.append(cartItemEl);
+    }
+    
+    function _displayCart() {
+        const cartLength = cartItemsList.lineItems.length;
+        const cartContainer = document.querySelector(".cart");
+        const cartHeading = createDomNode("h2",undefined,"Cart");
```

Comment:
To avoid passing explicit undefined in such cases, it's always a good idea to convert the arguments to named parameters for >=3 parameters

If you did that, you'd just have to pass `{ type: "h2", tectContent: "Cart"}`